src/Utilities/Algorithm.hpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/*!
+ * \ingroup Utilities
+ * These functions are constexpr reimplementations of some standard library
+ * functions, designed to work with cpp17::array.
+ */
+
+#pragma once
+
+#include <functional>
+#include <iterator>
+
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace cpp20 {
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::reverse that is constexpr, for bidirectional
+ * iterators; taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/algorithm
+ */
+template <class BidirIt>
+constexpr void reverse(BidirIt first, BidirIt last,
+                       std::bidirectional_iterator_tag) {
+  while (first != last) {
+    if (first == --last) {
+      break;
+    }
+    cpp20::iter_swap(first, last);
+    ++first;
+  }
+}
+
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::reverse that is constexpr, for random access
+ * iterators; taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/algorithm
+ */
+template <class RandomAccessIterator>
+constexpr void reverse(RandomAccessIterator first, RandomAccessIterator last,
+                       std::random_access_iterator_tag) {
+  if (first != last) {
+    for (; first < --last; ++first) {
+      cpp20::iter_swap(first, last);
+    }
+  }
+}
+
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::reverse that is constexpr;
+ * taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/algorithm
+ */
+template <class BidirectionalIterator>
+constexpr void reverse(BidirectionalIterator first,
+                       BidirectionalIterator last) {
+  cpp20::reverse(first, last,
+                 typename std::iterator_traits<
+                     BidirectionalIterator>::iterator_category());
+}
+
+/*!
+ * \ingroup Utilities
```

Comment:
Have you checked the documentation?

Reply:
I did check, and the docs looked fine to me. Specific issue I should double check?

Reply:
I don't see how this can show up in the `UtilitiesGroup` since the name is wrong.